extensions/wikia/AdEngine/js/template/modalOasisHandler.js

```diff
@@ -27,11 +28,30 @@ define('ext.wikia.adEngine.template.modalOasisHandler', [
 				modal.$content.append(adContainer);
 				modal.$element.width('auto');
 
+				function timer() {
+					closeButtonDelay = closeButtonDelay - 1;
+
+					if (closeButtonDelay < 0) {
+						clearInterval(counter);
+						$counter.hide();
+						modal.$close.show();
+						return;
+					}
+
+					$counter.text(closeButtonDelay);
+				}
+
 				if (closeBtnDelay > 0) {
+					var $header = modal.$close.parent(),
+						$counter = $(document.createElement('div'));
+
 					modal.$close.hide();
-					setTimeout(function(){
-						modal.$close.show();
-					}, closeBtnDelay);
+
+					$counter.addClass('close-counter');
+					$counter.text(closeButtonDelay);
+					$header.prepend($counter);
+
+					var counter = setInterval(timer, 1000);
```

Comment:
Move the declaration to the top of delta function and here just leave the assignment, please.